### ISSUE TITLE
Integer Overflow

### ISSUE THREAD

**Issue by DarioS** [MarioniLab/scran#115]
I have a typical-sized data set and would not expect any numerical issues. But it happens. Latest release versions of R and packages.

```r
library(SingleCellExperiment)
library(scuttle)
library(scran)
SCE <- SingleCellExperiment(assays = list(counts = matrix(rpois(21711 * 111671, 10), ncol = 111671)))
SCE <- logNormCounts(SCE)
SCE$cellType <- rep(c("Fibroblasts", "Other", "Tumor cells"), c(40967, 7374, 63330))
scoreMarkers(SCE, SCE$cellType, lfc = 3)
Warning messages:
1: In left.ncells * right.ncells : NAs produced by integer overflow
2: In left.ncells * right.ncells : NAs produced by integer overflow
```

The dimensions of the matrix and number of samples per class are identical to a biological data set.

**Comment by LTLA** [comment]
Oops. Fixed by f180b0fe767731aaa53f501a8431356cb599684c in 1.31.2. Check if it works and I'll push it to release.

**Comment by DarioS** [comment]
It was tried with biological data and the problem is gone.
